### PR TITLE
Fire scorches grass :O)

### PR DESCRIPTION
**Pull request by kugamo**
## About The Pull Request

Fire will now scorch grass if burning upon it! Grass will get scorched more, the hotter the fire and with the number of burns.

Only the gm grass for now. I couldnt test sorokyne grasses because the map kept crashing upon loading (dont ask me why).

credit to @carlarctg for the idea, very cool 😎 

## Why It's Good For The Game

Muh immersion. Its a cool interaction that adds to the environment, places fought on will look more the battleground all singed and scorched. 

https://user-images.githubusercontent.com/16618648/201831348-67786e67-658c-4872-9643-7ef2f4274276.mp4

## Changelog

:cl:
add: Fire now scorches grassy turfs!
/:cl:

### CODE REVIEW

**Comment by Huffie56** [comment]
Good stuff. Could you document a bit deeper on how it work? would be much appreciated.
I could see other dev may want to expand it to turn some part of beach into glass for example.
I would do it if i was better at code.
Also did you test it a bunch to see if it could cause performance issue(not an expert but i hate lag spike)
thanks for your contribution.

**Review by monkeysfist101 — APPROVED** [review]
Sprites approved.

**Review by fira — APPROVED** [review]
Looks OK overall. ~~I'm just not 100% sure about icon_state ordering, because from memory changing it will kill overlays~~ i'm dumb

File: code/game/turfs/open.dm [hunk 1]
```diff
@@ -65,12 +68,60 @@
...
 	. = ..()
 	var/ceiling_info = ceiling_desc(user)
 	if(ceiling_info)
 		. += ceiling_info
+	if(scorchedness)
+		switch(scorchedness)
+			if(1)
+				. += "Lightly Toasted."
+			if(2)
+				. += "Medium Roasted"
```
Comment (carlarctg) [inline code/game/turfs/open.dm:122]:
```suggestion
				. += "Medium Roasted."
```

File: code/game/turfs/open.dm [hunk 2]
```diff
@@ -65,12 +68,60 @@
...
+			if(3)
+				. += "Well Done"
```
Comment (carlarctg) [inline code/game/turfs/open.dm:124]:
```suggestion
				. += "Well Done."
```

**Review by fira — APPROVED** [review]

**Pull request merged**